profiler/profiler.go

```diff
@@ -226,15 +233,22 @@ func (p *profiler) collect(ticker <-chan time.Time) {
 			}
 
 			for _, t := range p.enabledProfileTypes() {
-				profs, err := p.runProfile(t)
-				if err != nil {
-					log.Error("Error getting %s profile: %v; skipping.", t, err)
-					p.cfg.statsd.Count("datadog.profiler.go.collect_error", 1, append(p.cfg.tags, t.Tag()), 1)
-					continue
-				}
-				for _, prof := range profs {
-					bat.addProfile(prof)
-				}
+				wg.Add(1)
+				go func(t ProfileType) {
+					defer wg.Done()
+					profs, err := p.runProfile(t)
+					if err != nil {
+						log.Error("Error getting %s profile: %v; skipping.", t, err)
+						p.cfg.statsd.Count("datadog.profiler.go.collect_error", 1, append(p.cfg.tags, t.Tag()), 1)
+					}
+					mu.Lock()
+					defer mu.Unlock()
+					completed = append(completed, profs...)
```

Comment:
🐞 `completed` growing unbound, not getting reset (found by nick)

testing nit: would be nice if we had a test that does multiple rounds of collect/upload to catch these kind of bugs in the future.

Reply:
Done, updated `TestAllUploaded` to check the second batch of profiles, which failed before fixing this bug.